extractor/docker/docker_test.go

```diff
@@ -404,76 +406,100 @@ func TestApplyLayers(t *testing.T) {
 }
 
 func TestExtractor_ExtractLayerFiles(t *testing.T) {
+	te, tr := setupDockerEnvironment()
+	defer te.Close()
+	defer tr.Close()
+
+	serverAddr := tr.Listener.Addr().String()
```

Comment:
Just curious, will the httptest server through test{engine,registry} be ready before the tests get run?

Reply:
I think httptest.NewServer waits until it can accept a request.
https://github.com/golang/go/blob/master/src/net/http/httptest/server.go#L304

Reply:
I meant to say if the server is not ready yet and the request is made prior to it being ready, the request would fail. You can see an example here https://github.com/aquasecurity/testdocker/blob/master/registry/server_test.go#L51-L58

Reply:
What do you mean? Your example uses goroutine and it is the cause. But I meant `httptest` waited until the server is ready unless it is called in goroutine. It means the next line will not be called and a request will not be sent until the server can accept a request.

I think this example will never fail.
https://play.golang.org/p/QDDTLsmMT2h

Reply:
I got it.